DNM: android changes

hyp opened the pull request:
for testing

rintaro committed:
[Macros] Use 'LibraryPluginProvider' in swift-plugin-server

Move 'PluginProvider' logic to 'SwiftLibraryPluginProvider' module
so the compiler can reuse that logic for the actual in-process plugins.

calda committed:
[SE-0365] Fix issue where implicit self was unexpectedly disallowed in nested closures

al45tair committed:
[Build] Add the new fully-static Linux SDK.

Declare a new `LINUX_STATIC` SDK and configure it.

Add options to set the build architectures for the `LINUX` and
`LINUX_STATIC` SDKs, similar to what we have for Darwin, because
we'll be cross-compiling.

Also add an option to point the build system at the sources for
the musl C library, which we're using for `LINUX_STATIC`.

rdar://123503470

al45tair committed:
[Platform] Update musl modulemap and improve musl platform support.

We need to declare separate top-level modules for every header in
the musl C library.  Note that our set-up here does require some
tweaks to musl itself (specifically, the `alltypes.h` header needs
some fairly serious massaging to make it module compatible; in
particular, we need to pull out every type declaration into its
own separate modularized header in `bits/types`).

Also update `Platform.swift` a little for more musl support.

rdar://123503615

al45tair committed:
[Runtime] Fix generation of .lnk files for static linking.

Use the new `SWIFT_SDK_<sdk>_STATIC_LINKING_SUPPORTED` and
`_STATIC_ONLY` flags instead of hardcoding support for Linux
and WASI.

Also, use the `_LIB_SUBDIR` variable rather than lowercasing
the SDK.

rdar://123504757

al45tair committed:
Add support to the compiler for musl and the LINUX_STATIC SDK.

This is really just about setting appropriate defaults (such as
making sure that the static Linux triple causes us to use lld).

rdar://123506306

al45tair committed:
[Backtracing] Make the backtracer work for fully-static Linux.

The backtracer and its build system need a couple of changes to make
them work with the new fully-static Linux support.

rdar://123507656

al45tair committed:
[Stdlib][Build] Fix the standard library build for fully static Linux.

Mostly this just means adding `Musl` as a module dependency of various
things and making sure that we build things for `swift_static` even
if `SWIFT_BUILD_STATIC_STDLIB` isn't enabled.

There's also a slight difference in the declaration of `memcmp()`;
musl's declaration is more like the one we have on Darwin.

rdar://123508245

al45tair committed:
[IRGen] Use llvm.used on ELF instead of llvm.compiler.used.

We changed to `llvm.compiler.used` because of the behaviour of `gold`,
which refuses to coalesce sections that have different `SHF_GNU_RETAIN`
flags, which causes problems with metadata.

Originally I thought we were going to have to generate two sections
with distinct names and have the runtime look for both of them, but
it turns out that the runtime only wants to see sections that have
`SHF_GNU_RETAIN` in any case.  It's really the reflection code that
is interested in being able to see non-retained sections.  The upshot
is that we don't need to use `llvm.compiler.used`; it's just fine if
we have duplicate sections, as long as the reflection code looks for
them when it's inspecting an ELF image.

This also means we no longer need to pass `-z nostart-stop-gc` to the
linker if we're using `lld`.

rdar://123504095

al45tair committed:
[Runtime] Tidy up section declarations slightly.

This also calls out the reflection sections, even though we don't
actually need to do anything special for them (as it turns out).

rdar://123504095

al45tair committed:
[RemoteInspection] Update ReflectionContext to cope with multiple sections.

When linking with `gold`, we can end up with two separate sections
for each of the reflection sections; this happens because `swiftrt.o`
declares them with `SHF_GNU_RETAIN` set (to stop them from vanishing
when section GC is enabled), but if the compiler is set to allow the
reflection information to be stripped then *it* will generate them
*without* `SHF_GNU_RETAIN`.

The other Linux linkers will coalesce the two sections and leave the
`SHF_GNU_RETAIN` flag set, but `gold` does not, and adds both of them
separately to the output.

The upshot is that we want `ReflectionContext` to look for both the
retained and non-retained sections and read the data from both of
them.

rdar://123504095

al45tair committed:
[IRGen] When creating a module object, mark `__Swift_AST` as used.

Without doing this, `__Swift_AST` gets stripped from the output.
We also need to call `IGM.finalize()` before `::performLLVM()` in the
`createSwiftModuleObjectFile()` function, so that we update the
section to mark it as retained.

rdar://123504095

al45tair committed:
[IRGen] Set `TargetOpts.UseInitArray` in `getIRTargetOptions()`.

IRGen sets up the `llvm::TargetOptions` by itself, rather than copying
them from the Clang instance (it has to do this, at present, because
Clang doesn't provide a way to get it to initialise one).

Unfortunately, it didn't set `UseInitArray`, which when linking with
GNU `ld` or `gold` is *fine* because those linkers automatically
convert `.ctors` and `.dtors` sections into `.init_array` and
`.fini_array` respectively.  *However*, `lld` does *not* do this
conversion, so when using `lld`, if the compiler generates a
constructor or destructor function, it won't be called(!)

The fix is to set `UseInitArray` properly; I chose to copy the setting
from Clang, just in case Clang knows of a reason why it shouldn't be
`true`.

rdar://123504095

al45tair committed:
[Test] Fix function_sections test to support different linkers.

`gold` and `lld` produce map files in different formats; this test
can and should work for both of them.

rdar://123504095

al45tair committed:
[Test] Add linker_overridden feature.

When testing different linkers, it's sometimes useful to run the tests
with `SWIFT_DRIVER_TEST_OPTIONS=" -use-ld=<linker>"`.  If we do this,
it will break a handful of tests because they expect the compiler driver
to choose an appropriate linker automatically.

To avoid having these fail, detect when someone has done this, and
set a new feature, `linker_overridden`, then mark the tests in question
with `UNSUPPORTED: linker_overridden`.

rdar://123504095

al45tair committed:
[RemoteInspection] Use `bool` instead of `!!`.

Since this is adjacent to a `!=` operator, explicitly using `bool` seems clearer.

Co-authored-by: Evan Wilde <etceterawilde@gmail.com>

al45tair committed:
[Test] Update section_asm to work on non-ELF platforms again.

On ELF platforms, the output will be slightly different to non-ELF
platforms.  The test should ideally run everywhere, however, so
we need to be able to distinguish these platforms by changing
the lit.cfg to add some extra variables.

rdar://123504095

al45tair committed:
[RemoteInspection] Add a big comment to ReflectionContext.h.

Add a comment to ReflectionContext.h explaining the situation regarding
`SHF_GNU_RETAIN` and the reflection metadata sections.

rdar://123504095

al45tair committed:
[Runtime] Add the `retain` attribute to make the backtracer work.

Without `retain` here, we might remove the reference that pulls in
the backtracing support code.

rdar://123504095

al45tair committed:
[Test][SourceKit] Use the *builder*'s swiftrt.o when in hosttools.

We were linking with the newly built `swiftrt.o` when in hosttools mode,
which is wrong because the newly built `swiftrt.o` does not match the
compiler we were using for the `SwiftCompilerSources`.

This manifests as a failure in

  `SwiftCompilerSources/Sources/Optimizer/Utilities/Verifier.swift`

because `self is ForwardingInstruction` fails as we can't find the
protocol conformance records.

rdar://123504095

al45tair committed:
[Runtime] Tidy up a couple of minor nits.

This doesn't change any code, just makes things look slightly
neater.

rdar://123504095

al45tair committed:
[RemoteInspection] Change "return false" to "return {}".

There are a number of instances in `ReflectionContext.h` where we
are doing `return false` with an `std::optional<...>` where it seems
we really mean to return an empty optional instead.

(The way to do this is either `return {}` or `return std::nullopt`.)

rdar://123504095

calda committed:
Fix edge cases related to nested autoclosures, invalid weak self unwrapping

calda committed:
Improve quality of 'capture self explicitly to enable implicit self in this closure' and 'variable other than self captured here under the name self does not enable implicit self' diagnostics

egorzhdan committed:
[cxx-interop] Faster `std::string` initialization from String

Instead of appending characters one-by-one, which resizes the resulting string multiple times, let's reserve the required number of bytes beforehand.

rdar://127423949

calda committed:
Look up the outer self decl via the capture's parent initializer instead of ASTScope::lookupSingleLocalDecl

calda committed:
Merge branch 'main' into cal--fix-70089

augusto2112 committed:
[DebugInfo] Always emit unsubsituted generic types with size 0

Unsubstituted generic types shouldn't have a size, precisely because
the generic parameters aren't substituted in. Always emit them with a
size 0.

hyp committed:
[android] squash all android changes

calda committed:
Always validate self capture

hyp commented:
@swift-ci please build toolchain

finagolfin commented:
Note that the toolchain build will not compile the Android libraries, so all this is good for is making sure your Android changes do not break the build for other platforms.

elsh committed:
Handle package exportability.

This PR treats package access level as exportable, preventing
internally imported types from accidentally being declared in
package decl signatures.

Added package-specific cases to ExportabilityReason and
DisallowedOriginKind to track the validity of imported types
at use sites with package access scope. Added tests to cover
variety of use cases.

Resolves rdar://117586046&125050064&124484388&124306642

al45tair committed:
[Build] Detect ld.gold version and prefer lld if gold is too old.

If we're on a system that has ld.gold 2.35 or earlier, we want to use
lld instead because otherwise we end up with duplicate sections in the
output.

rdar://123504095

nate-chandler committed:
[BitwiseCopyable] Promote to feature.

SE-0426 was accepted.

nate-chandler committed:
[BitwiseCopyable] Promote ConformanceSuppression.

The ability to suppress conformance to BitwiseCopyable was part of the
accepted SE-428.

nate-chandler committed:
[BitwiseCopyable] Don't infer for @sensitive.

nate-chandler committed:
[BitwiseCopyable] Remove underscore.

nate-chandler committed:
[BitwiseCopyable] Remove #ifs.

nate-chandler committed:
[BitwiseCopyable] Deprecate unbound overloads.

The unconstrained overloads of loadUnaligned and storeBytes were
deprecated in SE-0426.

calda committed:
Improve 'shouldOnlyWarn' logic

calda committed:
Update DeclContext::getInnermostClosureForSelfCapture to return a ClosureExpr

calda committed:
Simplify autoclosure check in shouldOnlyWarn

hyp commented:
> Note that the toolchain build will not compile the Android libraries, so all this is good for is making sure your Android changes do not break the build for other platforms.

Please take another look! I can now build the Android stdlib on linux.

hyp commented:
@swift-ci please build toolchain

finagolfin commented:
> I can now build the Android stdlib on linux.

I'm not sure what you're referring to, these CI presets don't enable building the Android stdlib or have the NDK installed to do it.

slavapestov committed:
SIL: Plumb pack element captures through type lowering

slavapestov committed:
SIL: Rename misleading getFunctionInterfaceTypeWithCaptures()

slavapestov committed:
SIL: Support element archetypes inside lowerCaptureContextParameters()

slavapestov committed:
SILGen: Tiny cleanup

egorzhdan committed:
[cxx-interop] Do not crash when calling a subscript with unnamed parameter

This fixes a crash in SILGen when calling a C++ subscript that has an unnamed parameter from Swift.

The parameters from a C++ `operator[]` get carried over to the synthesized Swift subscript. If the Swift parameter has no name, there is no way to refer to it in SIL. However, the synthesized subscript accessor needs to pass this parameter to C++.

This change makes sure that we give a name to the Swift parameter if there isn't already a name on the C++ side.

rdar://83163841

Snowy1803 committed:
[DebugInfo] Fix wrong scope being used in AllocBoxToStack

artemcm committed:
Revert "Revert "[Dependency Scanning] Use a standalone `DependencyScanningFilesystem`on the scanner, sharing a common `status` cache from Clang's dependency service cache""

slavapestov committed:
SIL: Introduce TypeConverter::getForwardingSubstitutionsForLowering()

slavapestov committed:
SIL: Allow setting SILFunction's forwarding substitution map

slavapestov committed:
SILGen: Use the correct GenericEnvironment when lowering functions with pack element captures

slavapestov committed:
RequirementMachine: Better assertion when root type parameter is invalid

Previously getReducedType() would dump debug info if we had an invalid
dependent member type, but an invalid root type parameter would cause
a segfault. Fix this.

Snowy1803 committed:
[DebugInfo] Fix LoadableByAddress losing debug info

LoadableByAddress was losing debug info, including at -Onone.
When converting a load-store pair to a copy_addr, the debug info
attached to the load was not salvaged.
Additionally, the wrong scope was being attached to other debug
values.

Snowy1803 committed:
[DebugInfo] Fix AllocStackHoisting losing debug info

AllocStackHoisting was losing debug info, including at -Onone.
When two alloc_stacks of the same type are merged, one of them
would lose their debug variable. It is now salvaged, with an added
debug_value.

This case was previously only handled for noncopyable types, it is
now done in all cases.

hamishknight committed:
[cmake] Only check HostCompatibilityLibs for bootstrapping builds

Locally I build debug builds without the standard library,
using a copy of the stdlib in my release build. This hit
a CMake error here since the `HostCompatibilityLibs` target
isn't defined. Update to only access it when doing a bootstrapping
build.

al45tair committed:
[Build] Updated following initial comments from Eric.

Change the `--linux[-static]-arch` option to `--linux[-static]-archs`, on the
basis that it supports multiple values.

Other tidying.

rdar://123503470

al45tair committed:
[Build] Fix a conflict resolution mistake.

Apparently I accidentally removed a `]` while resolving a conflict in
`build_script_invocation.py.`

rdar://123503470

al45tair committed:
[Build] Fix Python linting issues.

The Python linter was complaining about import orders.

rdar://123503470

al45tair committed:
[Build] Add options to the Python build script tests.

Apparently I'd failed to add a couple of lines here.

rdar://123503470

al45tair committed:
[Platform] Updated after review comments.

Added the _modules workaround to make the compiler pull in the correct
module when someone does `import stdc_assert` (for instance).

Fix various niggles.

rdar://123503615

hyp commented:
@swift-ci please build toolchain

rintaro committed:
[SourceKit] Don't use SourceEntitityWalker for placeholder expansion

Placeholder expansion should be a syntactic operation, but
`SourceEntityWalker` can invoke type checking operations, which causes
unexpected bahaviors including crashes.

rdar://121360941`

Snowy1803 committed:
Merge pull request #73387 from Snowy1803/allocbox-fix

[DebugInfo] Fix loss of variables at -Onone

etcwilde committed:
Build: EarlySwiftDriver: Use consistent Foundation/Dispatch

The early swift driver build would switch the Dispatch and Foundation
build used while building between an initial clean build and an
incremental rebuild. The early swift driver should consistently use the
same Foundation and Dispatch for each build.

This change tells the early swift driver stops build-script from telling
the early driver build to use the Foundation and Dispatch that it built.

egorzhdan committed:
Merge pull request #72268 from apple/egorzhdan/std-string-reserve

[cxx-interop] Faster `std::string` initialization from String

cachemeifyoucan committed:
[ScanDependencies] Fix assertion failure when failed to scan header

Fix an assertion failure when bridging header dependency scanning
failed.

rdar://127459045

elsh committed:
Update to ensure downgradeToWarning doesn't get reset

Snowy1803 committed:
[DebugInfo] Fix PhiExpansion losing debug info

When replacing a struct with its field, it now emits a debug_value
with a fragment, instead of dropping the debug_value.

slavapestov committed:
Merge pull request #73393 from slavapestov/pack-expansion-closures-part-3

SIL type lowering support for closures that capture pack element archetypes

drexin committed:
[IRGen] Fix alignment for imported C types in CVW

rdar://127279770

When an imported C type is over or under aligned, we did not use the alignment of the type, but computed the maximum alignment of its components, causing alignment issues in compact value witnesses.

nate-chandler committed:
[Test] Expose current pass.

Some clients need to access their parent transformation.

nate-chandler committed:
[Test] Added sil_combine_instruction.

nate-chandler committed:
[SIL] Expose indirect error type on convention.

nate-chandler committed:
[SILCombine] Handle indirect error in apply(conv).

Previously, only the indirect results and the parameters were handled.
However, the indirect error doesn't show up in the list of indirect
results (or in the list of parameters).  But it sure does show up in the
list of arguments.  Here, that argument too is handled.

rdar://127452206

jckarter committed:
SILGen: Correct handling of subscripts with addressors.

Remove improper special-case handling of subscripts in `findStorageReferenceExprForMoveOnly`.
The correct thing to do for any storage decl ref of noncopyable type is to emit it as a borrow
if it's implemented using storage, a read accessor, or an addressor. Fixes rdar://127335590.

etcwilde committed:
Merge pull request #73399 from etcwilde/ewilde/earlyswiftdriver-use-consistent-sdk

Build: EarlySwiftDriver: Use consistent Foundation/Dispatch

drexin committed:
Merge pull request #73405 from drexin/wip-127279770

[IRGen] Fix alignment for imported C types in CVW

al45tair committed:
[Build] Further static Linux SDK support changes.

Fix some indentation issues.

Change `build-script-impl` to make `build-linux-static` a positive argument.

Fix documentation for `--linux-archs` and `--linux-static-archs` (the options
are comma separated for `build-script`, but semicolon separated for
`build-script-impl`).

Set the default for `linux-static-archs` to `x86_64, aarch64` so that we
install the expected content in the toolchain.

Add missing default for `test_linux_static`.

Make sure to pass down `--skip-build-linux` and `--build-linux-static`.

Factor out config file generation and call it from the install step in `llvm.py`
as well as from the build step.

rdar://123503470

al45tair committed:
[Build] A couple more Python fixes.

Remove unused import of `importlib.resources`.

Set `test_linux_static` to `False` if tests are disabled.

rdar://123503470

al45tair committed:
[ClangImporter] Don't put built-in headers in system modules for musl.

We only need this for WASI and Glibc.

rdar://123506306

cachemeifyoucan committed:
Merge pull request #73402 from cachemeifyoucan/eng/PR-127459045

[ScanDependencies] Fix assertion failure when failed to scan header

rintaro committed:
Merge pull request #73394 from rintaro/sourcekit-expandplaceholder-rdar121360941

[SourceKit] Don't use SourceEntitityWalker for placeholder expansion

hyp commented:
@swift-ci please build toolchain

artemcm committed:
Merge pull request #73374 from apple/revert-73234-revert-72932-UseClangStatCacheForScanningFS

Revert "Revert "[Dependency Scanning] Use a standalone `DependencyScanningFilesystem`on the scanner, sharing a common `status` cache from Clang's dependency service cache""

DougGregor committed:
Bring back `TaskGroup.next()` and `ThrowingTaskGroup.next()` as public API

With the introduction of `next(isolation:)` into the task group types,
we removed the public APIs for the no-argument `next()` versions,
leaving them only as `@usableFromInline internal` entrypoints for the
ABI. Because the `next(isolation:)` versions have a default argument,
this was sufficient for providing source compatibility for calls, but
not for protocol conformances.

Bring these APIs back publicly. Fixes rdar://127499568.

DougGregor committed:
Disfavor the older `next()` APIs on TaskGroup

drexin committed:
[Runtime] Mask out spare bits after copying for Error in CVW

rdar://127379960

When the spare bits of an Error objects are used to store tag bits, this caused the enum tag to be lost, which caused the wrong enum cases to be matched.

gottesmm committed:
[region-isolation] Change diagnostics from using the term transferring -> sending.

rdar://127580781

jckarter committed:
Merge pull request #73410 from jckarter/noncopyable-subscript-borrow-fix

SILGen: Correct handling of subscripts with addressors.

augusto2112 committed:
Merge pull request #72445 from augusto2112/zero-unsub-types-main

[DebugInfo] Always emit unsubsituted generic types with size 0

DougGregor committed:
Update API expectations for (Throwing)TaskGroup.next()

elsh committed:
Merge pull request #73161 from apple/elsh/pkg-export

[Sema] Handle package exportability.

nate-chandler committed:
Merge pull request #73408 from nate-chandler/rdar127452206

[SILCombine] Handle indirect error in apply(convert_function).

hamishknight committed:
Merge pull request #73383 from hamishknight/fix-stdlibless-build

[cmake] Only check HostCompatibilityLibs for bootstrapping builds

gottesmm committed:
[region-isolation] If we can infer the callee's name, use that instead of just saying 'callee'.

I also wordsmithed the error message to use the term 'risk' instead of less
negative terms.

nate-chandler committed:
[SIL] Hollow out Builtin.copy, deprecate _copy.

The copy operator has been implemented and doesn't use it.  Remove
`Builtin.copy` and `_copy` as much as currently possible.

Source compatibility requires that `_copy` remain in the stdlib.  It is
deprecated here and just uses the copy operator.

Handling old swiftinterfaces requires that `Builtin.copy` be defined.
Redefine it here as a passthrough--SILGen machinery will produce the
necessary copy_addr.

rdar://127502242

jckarter committed:
Look through `@Sendable` conversions when considering C function pointer conversions.

Adding (or removing) `@Sendable` doesn't add context, so it shouldn't prevent
conversion to a C function pointer. Fix for rdar://127521718.

nate-chandler committed:
Merge pull request #73422 from nate-chandler/rdar127502242

[SIL] Hollow out Builtin.copy and deprecate _copy.

DougGregor committed:
Merge pull request #73425 from DougGregor/task-group-next-public

al45tair committed:
Merge pull request #72061 from al45tair/eng/PR-123504095

[Linux][Runtime][IRGen] Mark metadata sections as retained and support section GC.

stephentyrone committed:
Make static [milli/micro/nano]seconds members on Duration inlinable (#73429)

This means that they can't use _Int128 for their implementation, but efficient implementation of these using only 64b arithmetic is pretty straightforward, so that's OK. This allows them to be specialized and mostly optimized away in release builds.

stephentyrone committed:
Avoid using generic static members on Duration across module boundaries (#73419)

* Avoid using generic static members on Duration across module boundaries

Because the clocks are implemented in Concurrency, but Duration is in the Swift module, these don't get specialized. Add a fully-concrete internal init in Concurrency to avoid the problem.

* Call self.init(_high:low:) explicitly.

* Add availability annotation.

kateinoigakukun committed:
modulewrap: Disable ObjC interop by default on non-Darwin platforms

0a5653dbaf03c21e175d317a5344e446c64a4ef0 started to call
`IGM.finalize()`, which leads the Clang instance to emit ObjC metadata
sections when the ObjC interop is enabled. Emitting ObjC metadata
sections is not well supported on non-Darwin platforms and causes
crashes for WebAssembly and COFF object formats[^1].

modulewrap tool did not configure the ObjC interop option, so it always
enabled the ObjC interop. This patch aligns the default ObjC interop
value with other tools by disabling it on non-Darwin platforms.

[^1]: https://github.com/apple/llvm-project/blob/stable/20230725/clang/lib/CodeGen/CGObjCMac.cpp#L5068-L5074

calda committed:
Fix unexpected error by simplifying outermost capture validation

calda committed:
Refactor and simplify shouldOnlyWarn logic

nate-chandler committed:
Merge pull request #73235 from nate-chandler/bitwise-copyable/enable

[BitwiseCopyable] Promote to feature.

nate-chandler committed:
[AST] Remove Builtin.copy.

A vestigial remnant of it was left behind after
06921cfe84d3a5025d296e0a462db8a9ad8ac413 in order to avoid a reverse
condfail when building old swiftinterfaces that define

```swift
func _copy<T>(_ value: T) -> T {
  #if $BuiltinCopy
    Builtin.copy(value)
  #else
    value
  #endif
}
```

If the language feature is removed, though, such interfaces should again
be buildable because the branch where the language feature isn't defined
should be expanded.

rdar://127516085

calda committed:
Fix validation for async let thunks

calda committed:
Merge tag 'swift-DEVELOPMENT-SNAPSHOT-2024-05-01-a' of github.com:apple/swift into cal--fix-70089

Tag build swift-DEVELOPMENT-SNAPSHOT-2024-05-01-a

calda committed:
Remove unused validateOutermostCapture param

calda committed:
Fix outdated usage of isKnownNoEscape

al45tair committed:
[Build][Test] Fix TypeRoundTrip/round-trip.swift.

`TypeRoundTrip/round-trip.swift` tries to link with `libswiftRemoteInspection`,
but that library isn't in the default link path, so we need to add a `-L`
option to the command line in the test.

Also, we should check that `SWIFT_BUILD_REMOTE_MIRROR` is enabled and
disable the test if not, because otherwise `libswiftRemoteInspection` won't
have been built.

rdar://123503470

gottesmm committed:
[region-isolation] Only print the type of region that a value is in if it is not disconnected.

Just another diagnostic tweak.

gottesmm committed:
[region-isolation] Change terminology to use the term 'risk' instead of could.

gottesmm committed:
Fixes for clang importer tests that I missed

gottesmm committed:
Remove clang optimize off and loosen an error in the swift-version test.

gottesmm committed:
[region-isolation] Create a non-"callee" kind of error for async let.

I am doing this separately from the previous fix for just normal region
isolation since without transferring args and results enabled, we do not hit
this code path. But since I am here, I want to fix it at the same time.

rdar://127588005

gottesmm committed:
Merge pull request #73446 from gottesmm/rdar127580781

[region-isolation] Some diagnostic tweaks

gottesmm committed:
Merge pull request #73448 from gottesmm/pr-8d6637cbd2295148d54e0ee8c23ac999b4f7731a

[region-isolation] Create a non-"callee" kind of error for async let.

al45tair committed:
[Build] Only add -lfts for linux-static-*.

Looks like I'd unconditionally added -lfts to the Foundation build flags.

rdar://123503470

eeckstein committed:
cmake: set default value of BOOTSTRAPPING_MODE to HOSTTOOLS

This was already the intention before, but it didn't work because cmake's `option` only works for boolean variables.
Using `set` instead of `option` fixes that.

This change also enables SwiftCompilerSources in lldb on Windows.
https://github.com/apple/swift/issues/73322

eeckstein committed:
Force enabling SwiftCompilerSources in the build

This means that `bootstrapping=off` cannot be used anymore.

eeckstein committed:
tests: deprecate the `swift_in_compiler` lit feature

Now that SwiftCompilerSources are always enabled, `swift_in_compiler` is not needed anymore.
Set this feature unconditionally for tests which still include `REQUIRES: swift_in_compiler`.

egorzhdan committed:
Merge pull request #73337 from apple/egorzhdan/unnamed-subscript-param

[cxx-interop] Do not crash when calling a subscript with unnamed parameter

eeckstein committed:
SILVerifier: be more tolerant if errors were detected in diagnostic passes

SIL cannot be assumed to be 100% valid if errors were detected in diagnostic passes. For example in DefiniteInitialization.

Fixes some false verifier errors in case -sil-verify-all is used.

rdar://127519229
rdar://127518886

jckarter committed:
Merge pull request #73436 from jckarter/c-function-pointer-conversion-sendable

Look through `@Sendable` conversions when considering C function pointer conversions.

kateinoigakukun committed:
Merge pull request #73440 from kateinoigakukun/yt/objc-interop-modulewrap

modulewrap: Disable ObjC interop by default on non-Darwin platforms

rintaro committed:
Merge pull request #69236 from rintaro/macros-library-provider

[Macros] Use 'LibraryPluginProvider' in swift-plugin-server

al45tair committed:
Merge pull request #71839 from al45tair/eng/PR-123503470

[Build] Add the new fully-static Linux SDK.

aschwaighofer committed:
IRGen: memcpy instead of the outlined copy with take

The outlined copy with take would instantiate metadata to call the value
witness table's copy function.

rdar://126751753

Snowy1803 committed:
Merge pull request #73404 from Snowy1803/phixpansion

[DebugInfo] Fix PhiExpansion losing debug info

gottesmm committed:
[region-isolation] Some more diagnostic wordsmithing.

rdar://127580781

eeckstein committed:
Merge pull request #73453 from eeckstein/fix-verifier

SILVerifier: be more tolerant if errors were detected in diagnostic passes

slavapestov committed:
IRGen: Fix incorrect conformance check

Fixes rdar://126637139, rdar://126779977.

slavapestov committed:
IRGen: Fix some more incorrect conformance checks

nate-chandler committed:
Merge pull request #73443 from nate-chandler/rdar127516085

[AST] Remove Builtin.copy.

nate-chandler committed:
[Test] Reenable SIL unit tests on windows.

Use `fflush(stdout)` from C++.

Issue 73252.

drexin committed:
Merge pull request #73426 from drexin/wip-127379960

[Runtime] Mask out spare bits after copying for Error in CVW

ahoppen committed:
[Lexer] Fix an assertion failure when invoking code completion inside a single-quoted string literal

rdar://126924962

rintaro committed:
Revert "[Macros] Use 'LibraryPluginProvider' in swift-plugin-server"

etcwilde committed:
Merge pull request #73469 from apple/revert-69236-macros-library-provider

Revert "[Macros] Use 'LibraryPluginProvider' in swift-plugin-server"

drodriguez committed:
[test][Linux] Improve Gold/LLD detection for function_sections.swift

In #72061 the test was modified because LLD and Gold linker map formats
are different, so they need different CHECKs. The original method in
PR #72061 presents problems when the output is not exactly only one
line.

These changes use the first line of the linker map file to decide if the
Gold or the LLD linker map checks should be used, since the format
differs in both for the first line.

This way of checking works in my Linux machine when using the default
linker, when forcing Gold with `-use-ld=gold` and when forcing LLD with
`-use-ld=lld`.

hborla committed:
[Sema] Non-exhaustive switch statements are always an error in Swift 6.

gottesmm committed:
Merge pull request #73459 from gottesmm/pr-baa351cf701b5b9d9bd2568d678990c9b225bab0

[region-isolation] Some more diagnostic wordsmithing.

gottesmm committed:
[region-isolation] Some more diagnostic wordsmithing.

rdar://127580781

hborla committed:
[Features] Replace `-enable-nonfrozen-enum-exhaustivity-diagnostics` with an
upcoming feature.

The bespoke flag still works as a way to enable the `NonfrozenEnumExhaustivity`
upcoming feature. `NonfrozenEnumExhaustivity` is enabled by default in the
Swift 6 language mode as errors, and enabled by default in the Swift 5 language
mode as warnings.

gottesmm committed:
Merge pull request #73471 from gottesmm/pr-f33bb59b938e7dde6c15c1831bcd9afcc7c31654

[region-isolation] Some more diagnostic wordsmithing.

eeckstein committed:
Merge pull request #73417 from eeckstein/force-bootstrapping

Force enabling SwiftCompilerSources in the build

kateinoigakukun committed:
[Build] Repair the build for non-static-only static stdlib builds

When SWIFT_BUILD_STATIC_STDLIB=ON, SWIFT_BUILD_DYNAMIC_STDLIB=OFF, and
the sdk being built is not a static-only (e.g. WASI), the build fails
due to duplicate custom command rules against the same output path.
In the case of WASI, the static archive should be copied into lib/swift
by the first lipo target, and then the second lipo target should copy it
into lib/swift_static.

```
CMake Error at cmake/modules/SwiftAddCustomCommandTarget.cmake:144 (add_custom_command):
  Attempt to add a custom rule to output

    /home/build-user/build/buildbot_linux/wasmstdlib-linux-x86_64/lib/swift_static/wasi/libswiftCore.a.rule

  which already has a custom rule.
Call Stack (most recent call first):
  stdlib/cmake/modules/AddSwiftStdlib.cmake:673 (add_custom_command_target)
  stdlib/cmake/modules/AddSwiftStdlib.cmake:2657 (_add_swift_lipo_target)
  stdlib/public/core/CMakeLists.txt:401 (add_swift_target_library)
```

al45tair committed:
Merge pull request #73474 from kateinoigakukun/yt/fix-non-static-only-static-stdlib

[Build] Repair the build for non-static-only static stdlib builds

ktoso committed:
[TaskLocal] Carry access control modifiers into synthesized property (#73475)

ahoppen committed:
Merge pull request #73467 from ahoppen/assertion-failure-completing-in-single-quote-string-literal

[Lexer] Fix an assertion failure when invoking code completion inside a single-quoted string literal

slavapestov committed:
Merge pull request #70575 from calda/cal--fix-70089

[SE-0365] Fix issue where implicit self was unexpectedly disallowed in nested closures

hborla committed:
Merge pull request #73472 from hborla/exhaustive-switch-error

[Sema] Non-exhaustive switch statements are always an error in Swift 6.

nate-chandler committed:
Merge pull request #73460 from nate-chandler/gh73252

[Test] Reenable SIL unit tests on windows.

al45tair committed:
Merge pull request #71843 from al45tair/eng/PR-123504757

[Runtime] Fix generation of .lnk files for static linking.

al45tair committed:
Merge pull request #71844 from al45tair/eng/PR-123506306

Add support to the compiler for musl and the LINUX_STATIC SDK.

al45tair committed:
Merge pull request #71847 from al45tair/eng/PR-123507656

[Backtracing] Make the backtracer work for fully-static Linux.

al45tair committed:
Merge pull request #73470 from drodriguez/function-sections-fix

[Test][Linux] Improve Gold/LLD detection for function_sections.swift

al45tair committed:
[Platform] Rename modules for C library headers to match Darwin.

The Darwin headers have a naming convention for these modules; we should
match it.

rdar://123503615

aschwaighofer committed:
Merge pull request #73461 from aschwaighofer/optional_memcpy

IRGen: memcpy instead of the outlined copy with take

slavapestov committed:
Merge pull request #73457 from slavapestov/bitwise-copyable-lookup-conformance-irgen

IRGen: Fix incorrect BitwiseCopyable conformance lookups

DougGregor committed:
Revert "IRGen: memcpy instead of the outlined copy with take"

al45tair committed:
Merge pull request #71840 from al45tair/eng/PR-123503615

[Platform] Update musl modulemap and improve musl platform support.

al45tair committed:
Merge pull request #71848 from al45tair/eng/PR-123508245

[Stdlib][Build] Fix the standard library build for fully static Linux.

DougGregor committed:
Merge pull request #73507 from apple/revert-73461-optional_memcpy

Revert "IRGen: memcpy instead of the outlined copy with take"

jPaolantonio committed:
Remove duplicate hasAttachedPropertyWrapper check (#73456)

Co-authored-by: James Paolantonio <j_paolantonio@apple.com>

hyp committed:
Merge remote-tracking branch 'me/branch/android-complete' into HEAD
